src/WCSimLIGen.cc

```diff
@@ -0,0 +1,282 @@
+
+            G4cout << "Simulating events from " << injector["type"].get<string>() << " " << injector["idx"].get<string>() << G4endl;
+            injectorWavelength = injector["wavelength"].get<double>();
+            injectorPosition = injector["position"].get<vector<double>>();
+            injectorDirection = injector["direction"].get<vector<double>>();
+            injectorOffset = injector["offset"].get<double>();
+            thetaVals = injector["theta"].get<vector<double>>();
+            phiVals = injector["phi"].get<vector<double>>();
+            intensity = injector["intensity"].get<vector<double>>(); 
+            if ( photonMode ){
+                photonsFilename = injector["photonsfile"].get<string>();
+            }
+        }
+    
+    }	
+    
+    if ( photonMode ){
+        LoadPhotonList();
+    }
+    else {
+        LoadProfilePDF();
+    }
+}
+
+void WCSimLIGen::LoadPhotonList(){
+
+    string photonsFile = wcsimdir + photonsFilename;
+    cout << photonsFile << endl; 
+    ifstream photondata(photonsFile.c_str());
+    if ( ! photondata) {
+        cout << "WCSimLIGen [ERROR]: Failed to open " << photonsFilename << endl;
+    }
+    else {
+        cout << "WCSimLIGen photon list file: " << photonsFilename << " opened" << endl;
+    }
+
+    // Load in a csv list of x,y,z,px,py,pz in mm
+    string line;
+    // skip the first (header) line
+    getline(photondata,line);
+
+    while(getline(photondata,line)){
+        
+        Photon photon;
+        istringstream iss(line);
+        string value;
+        getline(iss,value, ',');
+        photon.x = stod(value);
+        getline(iss,value, ',');
+        photon.y = stod(value);
+        getline(iss,value, ',');
+        photon.z = stod(value);
+        getline(iss,value, ',');
+        photon.px = stod(value);
+        getline(iss,value, ',');
+        photon.py = stod(value);
+        getline(iss,value, ',');
+        photon.pz = stod(value);
+
+        myPhotons.push_back(photon);
+    }
+
+  photondata.close();
+
+}
+
+void WCSimLIGen::LoadProfilePDF(){
+
+    cout << "Reading the injector profile from file" << endl;
+
+    // Creates histogram of light injector profile
+    float thetaMin = thetaVals[0];
+    float phiMin = phiVals[0];
+    int nbins = thetaVals.size();
+    float thetaMax = thetaVals[nbins-1];
+    float phiMax = phiVals[nbins-1];
+
+    hProfile = new TH2D("hProfile","hProfile",nbins,thetaMin,thetaMax,nbins,phiMin,phiMax);
+    for (auto i=0u;i<thetaVals.size();i++){
+        hProfile->Fill(thetaVals[i],phiVals[i]);
+        int bin = hProfile->FindBin(thetaVals[i],phiVals[i]);
+        hProfile->SetBinContent(bin, intensity[i]);
+    }
+    cout << "Profile filled." << endl;
+}
+
+
+
+void WCSimLIGen::GeneratePhotons(G4Event* anEvent,G4int nphotons){
+
+    // Use particle gun to generate a point source
+    G4ParticleGun *myLIGun = new G4ParticleGun();
```

Comment:
Memory leak. Can you move this to `Initalise()` and rememeber to `delete` it in the destructor (or `Finalise()`)